komfydb/storage/heap_page.cc

```diff
@@ -14,34 +17,51 @@ absl::StatusOr<bool> TuplePresent(std::vector<uint8_t>& header, int i) {
   return header[i / 8] & (1 << (i % 8));
 }
 
+int ParseInt(uint8_t* data) {
+  int result = *reinterpret_cast<int*>(data);
+  return ntohl(result);
+}
+
 std::unique_ptr<IntField> ParseInt(std::vector<uint8_t>& data, int& data_idx) {
   std::unique_ptr<IntField> field =
-      std::make_unique<IntField>(*((int*)&data[data_idx]));
+      std::unique_ptr<IntField>(new IntField(ParseInt(data.data() + data_idx)));
   data_idx += Type::INT_LEN;
   return field;
 }
 
 std::unique_ptr<StringField> ParseString(std::vector<uint8_t>& data,
                                          int& data_idx) {
+  // TODO This function does not care about the length of string encoded
+  // in the data file and we parse the string until the first null byte.
+  // Should we leave it like this or read the amount of data specified by
+  // the integer before the string?
```

Comment:
I think - read only <length> bytes? That would make more sense if we have it on disk.

Reply:
Wdym "read only bytes"?

Reply:
"read the amount of data specified by the integer before the string"